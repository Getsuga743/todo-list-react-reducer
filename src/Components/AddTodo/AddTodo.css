.todo-form {
  z-index: 2;
  background-color: #ffffff;
  box-shadow: 0 14px 50px hsla(235, 19%, 35%, 0.205);
}
.todo-input {
  color: hsl(235, 19%, 35%);
  background-color: #ffffff;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  text-align: center;
  align-items: center;
  padding: 1rem 0;
  border: none;
  border-bottom: 0.5px solid hsl(233, 11%, 84%);
  width: 100%;
  font-size: 1.5rem;
}
.buttons {
  color: hsl(235, 19%, 35%);
  display: flex;
  flex-direction: row;
  justify-content: center;
  text-align: center;
  align-items: center;
  padding: 0;
  width: 100%;

  font-size: 1.5rem;
  border: none;
}
.buttons.todo-count {
  font-size: 1rem;
  align-items: center;
  word-spacing: 0.1rem;
  display: flex;
  width: 100%;
  height: 3rem;
}
.buttons .btn {
  border: none;
  width: 100%;
  height: 3rem;
  margin: 0;
  background-color: #ffff;
  font-family: 'Josefin Sans', sans-serif;
  color: hsl(235, 19%, 35%);
  border-left: 0.5px solid hsl(233, 11%, 84%);
}
