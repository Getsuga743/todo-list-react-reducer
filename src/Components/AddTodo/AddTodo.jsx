/* eslint-disable react/prop-types */
import React from 'react';
import hour from '../../helpers/getHour';
import useForm from '../../hooks/useForm';
import './AddTodo.css';

const AddTodo = ({ dispatch, count }) => {
  const [{ description }, handleInputChange, reset] = useForm({
    description: '',
  });
  const handleSubmit = (e) => {
    e.preventDefault();
    if (description.trim().length <= 1) {
      return;
    }
    const action = {
      type: 'add',
      payload: {
        id: new Date().getTime(),
        description: document.querySelector('#todoInput').value,
        done: false,
        date: `${new Date().getDate()}/${new Date().getMonth()}/${new Date().getFullYear()}`,
        // eslint-disable-next-line new-cap
        time: hour(),
      },
    };
    dispatch(action);
    reset();
  };
  const handleClear = (e) => {
    e.preventDefault();
    const action = { type: 'clear' };
    dispatch(action);
  };
  const handleKeyPress = (e) => {
    if (e && e.keyCode === 13) {
      document.querySelector('.todo-form').submit();
    }
  };

  return (
    <>
      <form name="todos" className="todo-form" onSubmit={handleSubmit}>
        <label htmlFor="todos" className="todo-label">
          <input
            className="todo-input"
            id="todoInput"
            type="text"
            name="description"
            placeholder="Insert todo"
            autoComplete="off"
            value={description}
            onChange={handleInputChange}
            onKeyPress={handleKeyPress}
          />
          <div className="buttons">
            <span className="buttons todo-count">
              {`${count} items left`}
            </span>
            <button className="btn btn-add" type="submit">
              Add
            </button>
            <button
              className="btn btn-clear"
              type="button"
              onClick={handleClear}
            >
              Clear Completed
            </button>
          </div>
        </label>
      </form>
    </>
  );
};

export default AddTodo;
