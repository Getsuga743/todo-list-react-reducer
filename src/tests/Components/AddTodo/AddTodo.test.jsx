/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import AddTodo from '../../../Components/AddTodo/AddTodo';
import demoTodos from '../../../fixtures/demoTodos';

describe('Pruebas en <TodoAdd />', () => {
  const dispatch = jest.fn();

  const wrapper = shallow(
    <AddTodo dispatch={dispatch} count={demoTodos.length} />,
  );
  test('debe de coincididir con el snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('No debe de llamar dispatch', () => {
    const formSubmit = wrapper.find('form').prop('onSubmit');
    formSubmit({ preventDefault() {} });
    expect(dispatch).toHaveBeenCalledTimes(0);
  });
  test('debe de llamar a la funcion dispatch', () => {
    const value = 'Aprender React';
    wrapper.find('input').simulate('change', {
      target: {
        value,
        name: 'description',
      },
    });

    // const formSubmit = wrapper
    //   .find('input')
    //   .simulate('keypress', { key: 'Enter' });
    // formSubmit({ preventDefault() {} });
    // expect(dispatch).toHaveBeenCalledWith({
    //   id: expect.any(Number),
    //   desc: value,
    //   done: false,
    // });
  });
});
