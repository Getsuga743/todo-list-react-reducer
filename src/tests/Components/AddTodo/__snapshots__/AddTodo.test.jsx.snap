// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoAdd /> debe de coincididir con el snapshot 1`] = `
<Fragment>
  <form
    className="todo-form"
    name="todos"
    onSubmit={[Function]}
  >
    <label
      className="todo-label"
      htmlFor="todos"
    >
      <input
        autoComplete="off"
        className="todo-input"
        id="todoInput"
        name="description"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Insert todo"
        type="text"
        value=""
      />
      <div
        className="buttons"
      >
        <span
          className="buttons todo-count"
        >
          2 items left
        </span>
        <button
          className="btn btn-add"
          type="submit"
        >
          Add
        </button>
        <button
          className="btn btn-clear"
          onClick={[Function]}
          type="button"
        >
          Clear Completed
        </button>
      </div>
    </label>
  </form>
</Fragment>
`;
